src/PolyType/SourceGenModel/SourceGenEnumerableTypeShape.cs

```diff
@@ -15,13 +15,13 @@ public sealed class SourceGenEnumerableTypeShape<TEnumerable, TElement> : Source
     public required ITypeShape<TElement> ElementType { get; init; }
 
     /// <inheritdoc/>
-    public required int Rank { get; init; }
+    public int Rank { get; init; }
```

Comment:
Let's reinstate this property.
```suggestion
    public required int Rank { get; init; }
```